main.go

```diff
@@ -98,14 +97,17 @@ func main() {
 	}
 
 	watchNamespace, ok := os.LookupEnv("WATCH_NAMESPACE")
-	if ok {
+	// We can't setup managerOptions.Cache.DefaultNamespaces = map[cache.AllNamespaces]cache.Config{} due to
+	// https://github.com/kubernetes-sigs/controller-runtime/issues/2628
+	if ok && watchNamespace != "nil" {
```

Comment:
Magic string value "nil" is _really_ bad choice, it looks like null but doesn't.
Empty string should work fine here.

Constant "AllNamespaces" is exactly empty string because it is invalid namespace name and natural "null" value for strings. While "nil" is odd, but still valid name.

Reply:
Okey, will change it to ""